File: Command/Go/command/Invoker.go
package command

type Invoker struct {
  command command
}

func (this *Invoker) SetCommand(cmd command) {
  this.command = cmd
}

func (this *Invoker) Run() string {
  return this.command.Excute()
}

func (this *Invoker) Undo() string {
  return this.command.Undo()
}

File: Command/PHP/Invoker.php
<?php
  class Invoker {
    protected $command;

    public function setCommand($cmd) {
      $this->command = $cmd;
    }

    public function run() {
      $this->command->excute();
    }

    public function undo() {
      $this->command->undo();
    }
  }
?>
